module/zfs/txg.c

```diff
@@ -689,9 +689,39 @@ txg_wait_synced(dsl_pool_t *dp, uint64_t txg)
 		    "tx_synced=%llu waiting=%llu dp=%p\n",
 		    tx->tx_synced_txg, tx->tx_sync_txg_waiting, dp);
 		cv_broadcast(&tx->tx_sync_more_cv);
-		cv_wait(&tx->tx_sync_done_cv, &tx->tx_sync_lock);
+		if (wait_sig) {
+			/*
+			 * Condition wait here but stop if the thread receives a
+			 * signal. The caller may call txg_wait_synced*() again
+			 * to resume waiting for this txg.
+			 */
+			if (cv_wait_sig(&tx->tx_sync_done_cv,
```

Comment:
The build failures are caused by the ZoL version of `cv_wait_sig()` returning `void` instead of `int`.  It would be best to update ZoL's `cv_wait_sig()` implementation to conform to the expected behavior.  That is to return zero when signaled and non-zero when the condition occurs.